Makefile

```diff
@@ -493,7 +499,11 @@ lint-docs:
 
 .PHONY: publish-docs
 publish-docs: lint-docs
-	mkdocs gh-deploy
+	docker run ${MKDOCS_RUN_ARGS} --rm -it \
+		-v ~/.ssh:/root/.ssh \
+		-v ${CURRENT_DIR}:/docs \
+		-v ~/.gitconfig:/root/.gitconfig \
+		${MKDOCS_DOCKER_IMAGE} gh-deploy -r ${GIT_REMOTE_REPO}
```

Comment:
This target is deprecated for a while now, we should remove it all together. This would also allow us to get rid of the hacks required for the necessity of git and ssh within the Docker container.